### ISSUE TITLE
fix: remove obsolete mcp[server] extra from dependencies

### ISSUE THREAD

**Issue by MarcusJellinghaus** [MarcusJellinghaus/mcp-code-checker#147]
Installing the package emits this warning:

```
warning: The package `mcp==1.27.0` does not have an extra named `server`
```

The `mcp[server]` extra no longer exists in modern `mcp` releases. The plain `mcp` package already provides `mcp.server.fastmcp`, which is what this project uses (see `src/mcp_tools_py/server.py`).

### Fix

In `pyproject.toml`, remove the redundant line:

```toml
"mcp[server]>=1.3.0",
```

Keep `"mcp>=1.3.0"` and `"mcp[cli]>=1.3.0"` unchanged.

### Acceptance Criteria

- [ ] `"mcp[server]>=1.3.0"` line removed from `pyproject.toml` dependencies
- [ ] `"mcp>=1.3.0"` and `"mcp[cli]>=1.3.0"` remain unchanged
- [ ] Installation no longer emits the `does not have an extra named server` warning

Tracked from MarcusJellinghaus/mcp_coder#702.
